tangelo/toolboxes/post_processing/histogram.py

```diff
@@ -0,0 +1,225 @@
+        lengths_bitstrings = set([len(k) for k in outcomes.keys()])
+        if len(lengths_bitstrings) > 1:
+            raise ValueError(f"Entries in outcomes dictionary not of consistent length. Please check your input.")
+
+        if n_shots > 0:
+            # Flags histograms whose frequencies do not add to 1.
+            sum_values = sum(outcomes.values())
+            if abs(sum_values-1) > epsilon:
+                raise ValueError(f"Histogram frequencies not summing very close to 1. (sum = {sum_values}).")
+
+            outcomes = {k: round(v*n_shots) for k, v in outcomes.items()}
+        elif n_shots < 0:
+            raise ValueError(f"The number of shots provided ({n_shots}) must be a positive value.")
+
+        self.counts = outcomes.copy()
+
+        if msq_first:
+            self.counts = {k[::-1]: v for k, v in self.counts.items()}
+
+    def __repr__(self):
+        """Output the string representation."""
+        return f"{self.counts}"
+
+    def __eq__(self, other):
+        """Two histograms are equal if they have same counts, i.e. same
+        bistrings and outcomes.
+        """
+        return (self.counts == other.counts)
+
+    def __add__(self, other):
+        """The aggregate_histograms function is used to add two Histogram objects."""
+        return aggregate_histograms(self, other)
+
+    def __iadd__(self, other):
+        new_histogram = self + other
+        self.__dict__ = new_histogram.__dict__
+        return self
+
+    @property
+    def n_shots(self):
+        """Return the number of shots encoded in the Histogram.
+
+        Returns:
+            int: Self-explanatory.
+        """
+        return sum(self.counts.values())
+
+    @property
+    def n_qubits(self):
+        """The length of the bistrings represents the number of qubits."
+
+        Returns:
+            int: Self-explanatory.
+        """
+        return len(list(self.counts.keys())[0])
+
+    @property
+    def frequencies(self):
+        """The frequencies are normalized counts vs the number of shots.
+
+        Returns:
+            dict: Frequencies in a {bistring: probability, ...} format.
+        """
+        return {bistring: counts/self.n_shots for bistring, counts in self.counts.items()}
+
+    def remove_bits(self, *indices):
+        """Method to remove bits on given indices.
+
+        Args:
+            indices (variable number of int): Qubit indices to remove.
+        """
+        new_counts = dict()
+        for bitstring, counts in self.counts.items():
+            new_bistring = "".join([bitstring[qubit_i] for qubit_i in range(len(bitstring)) if qubit_i not in indices])
+
+            if new_bistring in new_counts:
+                new_counts[new_bistring] += counts
+            else:
+                new_counts[new_bistring] = counts
+
+        self.counts = new_counts
+
+    def post_select(self, expected_outcomes):
+        """Method to apply post selection on Histogram data, based on a post
```

Comment:
Post-selection may be unambiguous for you, but for me it is somewhat. Should we include a link or mention this explicitly done on desired outcomes for specific qubit indices ?